### PR TITLE
Add bultins: i32/i64/f32/f64.eq/ne

### PR DESCRIPTION
**Pull request by technohippy**
⯈add eq/ne to builtin.ts

Related issue: https://github.com/AssemblyScript/assemblyscript/issues/1310

- [x] I've read the contributing guidelines
- [x] I've added my name and email to the NOTICE file

### CODE REVIEW

File: src/builtins.ts [hunk 1]
```diff
@@ -6565,6 +6686,78 @@ function builtin_f64_div(ctx: BuiltinContext): ExpressionRef {
...
+  ctx.contextualType = Type.i32;
+  return builtin_ne(ctx);
+}
+builtins.set(BuiltinNames.i32_ne, builtin_i32_ne);
+
+// i64.ne -> ne<i64>
+function builtin_i64_ne(ctx: BuiltinContext): ExpressionRef {
+  checkTypeAbsent(ctx);
+  ctx.typeArguments = [ Type.i64 ];
+  ctx.contextualType = Type.i64;
```
Comment (dcodeIO) [inline src/builtins.ts:6738]: The result of `i64.ne` is
```suggestion
  ctx.contextualType = Type.i32;
```

File: src/builtins.ts [hunk 2]
```diff
@@ -6565,6 +6686,78 @@ function builtin_f64_div(ctx: BuiltinContext): ExpressionRef {
...
+  ctx.contextualType = Type.i32;
+  return builtin_ne(ctx);
+}
+builtins.set(BuiltinNames.f32_ne, builtin_f32_ne);
+
+// f64.ne-> ne<f64>
+function builtin_f64_ne(ctx: BuiltinContext): ExpressionRef {
+  checkTypeAbsent(ctx);
+  ctx.typeArguments = [ Type.f64 ];
+  ctx.contextualType = Type.i64;
```
Comment (dcodeIO) [inline src/builtins.ts:6756]:
```suggestion
  ctx.contextualType = Type.i32;
```

File: std/assembly/builtins.ts [hunk 3]
```diff
@@ -136,6 +136,14 @@ export declare function mul<T>(left: T, right: T): T;
...
 export declare function div<T>(left: T, right: T): T;
 
+// @ts-ignore: decorator
+@builtin
+export declare function eq<T>(left: T, right: T): i32;
+
+// @ts-ignore: decorator
+@builtin
+export declare function ne<T>(left: T, right: T): i32;
+
```
[the remark this answers is not in the document]
Comment (dcodeIO) [reply std/assembly/builtins.ts:146]: I guess this also implies that adding the remaining tests would be good alongside :)

**Comment by technohippy** [comment]
Sorry,the history was broken. I'll make another pull request.

File: std/assembly/builtins.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (technohippy) [reply std/assembly/builtins.ts:146]: I fixed them on another pull req.
https://github.com/AssemblyScript/assemblyscript/pull/2258

**Pull request closed without merging**